src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs

```diff
@@ -581,17 +575,17 @@ public static void LogCritical([NotNull] this ILogger logger, int eventId, strin
             ILogValues state,
             Exception exception = null)
         {
-            logger.Log(logLevel, eventId, state, exception, _logValuesFormatter);
+            logger.Log(logLevel, eventId, state, exception, MessageFormatter);
         }
 
-        private static string LogValuesFormatter(ILogValues state, Exception exception)
+        private static string MessageFormatter(object state, Exception exception)
         {
             if (exception == null)
             {
                 return state?.ToString();
             }
 
-            return state?.ToString() + Environment.NewLine + exception;
+            return string.Format(CultureInfo.CurrentCulture, "{0}{1}{2}", state, Environment.NewLine, exception);
```

Comment:
good question...state being null is like the log message being null...yeah, a log statement without a message is unusual but definitely possible(scenario being: an exception is caught and user just wants to log the exception and does not want to provide any message)...I was initially thinking of why not have NotNull attributes on all the extensions where we accept `state`, but the above scenario invalidates it... @Eilon, thoughts?